src/Voron/Impl/EncryptionBuffersPool.cs

```diff
@@ -32,31 +33,32 @@ public unsafe class EncryptionBuffersPool : ILowMemoryHandler
 
         public EncryptionBuffersPool()
         {
-            var numberOfSlots = Bits.MostSignificantBit(_maxBufferSizeToKeepInBytes) + 1;
-            _items = new ConcurrentStack<NativeAllocation>[numberOfSlots];
+            var numberOfSlots = Bits.MostSignificantBit(MaxNumberOfPagesToCache * Constants.Storage.PageSize) + 1;
+            _items = new PerCoreContainer<NativeAllocation>[numberOfSlots];
 
             for (int i = 0; i < _items.Length; i++)
             {
-                _items[i] = new ConcurrentStack<NativeAllocation>();
+                _items[i] = new PerCoreContainer<NativeAllocation>();
             }
 
             LowMemoryNotification.Instance.RegisterLowMemoryHandler(this);
 
             _cleanupTimer = new Timer(CleanupTimer, null, TimeSpan.FromMinutes(1), TimeSpan.FromMinutes(1));
         }
 
-        public byte* Get(int size, out NativeMemory.ThreadStats thread)
+        public byte* Get(int numberOfPages, out int size, out NativeMemory.ThreadStats thread)
         {
-            size = Bits.PowerOf2(size);
+            numberOfPages = Bits.PowerOf2(numberOfPages); ;
+            size = numberOfPages * Constants.Storage.PageSize;
 
-            if (Disabled || size > _maxBufferSizeToKeepInBytes)
+            if (Disabled || numberOfPages > MaxNumberOfPagesToCache)
```

Comment:
Even here you will be over-allocating alot, just like before.
let's say you requested 275,000 pages (~2.1GB), next power of 2 of that is 2^19 = 524,288 pages which is 4GB of size.
I guess you need to keep the original numberOfPages..

Reply:
Sigh, sorry, that is me being silly. I wanted to avoid power of 2 if the size is greater than > 1MB